pkg/operation/care/checker.go

```diff
@@ -229,31 +229,35 @@ func (b *HealthChecker) checkNodes(condition gardencorev1beta1.Condition, nodes
 
 // CheckManagedResource checks the conditions of the given managed resource and reflects the state in the returned condition.
 func (b *HealthChecker) CheckManagedResource(condition gardencorev1beta1.Condition, mr *resourcesv1alpha1.ManagedResource) *gardencorev1beta1.Condition {
+	typeToFailedStatusToProcess := map[gardencorev1beta1.ConditionType]gardencorev1beta1.ConditionStatus{
+		resourcesv1alpha1.ResourcesApplied: gardencorev1beta1.ConditionFalse,
+		resourcesv1alpha1.ResourcesHealthy: gardencorev1beta1.ConditionFalse,
+	}
+
+	return b.checkManagedResourceForFailedConditions(condition, mr, typeToFailedStatusToProcess)
+}
+
+func (b *HealthChecker) checkManagedResourceForFailedConditions(condition gardencorev1beta1.Condition, mr *resourcesv1alpha1.ManagedResource, typeToFailedStatusToProcess map[gardencorev1beta1.ConditionType]gardencorev1beta1.ConditionStatus) *gardencorev1beta1.Condition {
 	if mr.Generation != mr.Status.ObservedGeneration {
 		c := b.FailedCondition(condition, gardencorev1beta1.OutdatedStatusError, fmt.Sprintf("observed generation of managed resource '%s/%s' outdated (%d/%d)", mr.Namespace, mr.Name, mr.Status.ObservedGeneration, mr.Generation))
 		return &c
 	}
 
-	toProcess := map[gardencorev1beta1.ConditionType]struct{}{
-		resourcesv1alpha1.ResourcesApplied: {},
-		resourcesv1alpha1.ResourcesHealthy: {},
-	}
-
 	for _, cond := range mr.Status.Conditions {
-		_, ok := toProcess[cond.Type]
+		failedStatus, ok := typeToFailedStatusToProcess[cond.Type]
 		if !ok {
 			continue
 		}
-		if cond.Status == gardencorev1beta1.ConditionFalse {
+		if cond.Status == failedStatus {
```

Comment:
As discussed, let's include in this PR also the check for `Unknown` condition status.

Reply:
With @plkokanov we were discussing what to do when the MR condition status is `Unknown`. What would be the reflection on the Seed/Shoot condition status in this case - `Unknown` or `False`? @rfranzke do you have any opinions?

Reply:
I would probably set it to `False` since we were able to check the MR, however it didn't report healthiness.

Reply:
Ok, thanks, then let's do it like this!